### PR TITLE
moving push templates to resources/reference and updated for ios

### PR DESCRIPTION
**Pull request by sbenedicadb**
## Description

In the future, a majority of the push templates documentation will be applicable for both Messaging and Campaign Classic extensions.  This PR re-organizes the presentation of these docs to be in a common location (`resources/reference`) and also adds documentation for the upcoming iOS push templates release.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/resources/reference/push-notifications/ios/configure-extension.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    class NotificationViewController: AEPNotificationViewController {
+        
+    }
+    ```
+
+1. Update the `Info.plist` for your Notification Content extension with the following values:
+
+    | Key | Type | Value |
+    | --- | --- | --- |
+    | `NSExtension.NSExtensionPrincipalClass` | `String` | `$(PRODUCT_MODULE_NAME).NotificationViewController` |
```
Comment (PravinPK) [inline src/pages/resources/reference/push-notifications/ios/configure-extension.md:61]: I found keeping `NSExtension.NSExtensionPrincipalClass` as `AEPNotificationContent. AEPNotificationViewController` works just fine. 

We can remove the first two steps here and make it simple.
Also, in this case we need to change the `configurePlist.png` image

File: src/pages/resources/reference/push-notifications/ios/create-extension.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    <img src="./assets/createNotificationContentExtension.png" />
+
+1. Click `Next`
+
+1. Name your Notification Content app extension
+
+1. Click `Finish`
+
+For more information about Notification Content app extensions, visit the [Apple documentation](https://developer.apple.com/documentation/usernotificationsui/customizing-the-appearance-of-notifications?language=objc).
```
Comment (PravinPK) [inline src/pages/resources/reference/push-notifications/ios/create-extension.md:19]: nit: we can remove the language parameter `language=objc` and have the site choose the default language

File: src/pages/resources/reference/push-notifications/templates/tabs/timer.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+| Sticky | ⛔️ | `adb_sticky` | string | Boolean value represented as a string "true" or "false".<br />When set to false or unset, the notification is automatically dismissed when the user clicks it in the panel. When set to true, the notification persists even when the user clicks it. |
+| Ticker | ⛔️ | `adb_ticker` | string | Sets the "ticker" text, which is sent to accessibility services. Prior to API level 21 (Lollipop), sets the text that is displayed in the status bar when the notification first arrives. |
+| Color - Small Icon | ⛔️ | `adb_clr_icon` | string | Color for the notification's small icon.<br />Represented as six character hex, e.g. `00FF00` |
+| Color - Background | ⛔️ | `adb_clr_bg` | string | Color for notification's background.<br />Represented as six character hex, e.g. `00FF00` |
+
+## Example
+
+Below is a sample of what a payload might look like for a carousel notification:
```
Comment (PravinPK) [inline src/pages/resources/reference/push-notifications/templates/tabs/timer.md:40]: `carousel notification`  to `timer notification`
Below looks like a carousel template payload. Needs to be Timer template.

**Review by PravinPK — CHANGES_REQUESTED** [review]

File: src/pages/resources/reference/push-notifications/ios/validate-integration.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        },
+        "adb_version" : "1.0",
+        "adb_template_type" : "basic",
+        "adb_title_ex" : "This is the expanded title of the notification",
+        "adb_body_ex": "This is the expanded body of the notification. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Pharetra magna ac placerat vestibulum lectus mauris ultrices eros in. Sit amet consectetur adipiscing elit ut aliquam purus sit amet. Orci a scelerisque purus semper. Laoreet id donec ultrices tincidunt.",
+        "adb_media" : "https://business.adobe.com/blog/fragments/products/experience-cloud/media_14480ef26514d00e5c224720f93790cb0b4499407.png?width=1000&height=500&format=png&optimize=medium"
+    }
+    ```
+
+1. Replace the text `[MY_BUNDLE_IDENTIFIER]` with the bundle identifier for your app. For example, `com.my.awesomeApp`
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/ios/validate-integration.md:37]: Missing period at the end of the sentence.

File: src/pages/resources/reference/push-notifications/templates/basic.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+---
+
+import Tabs from './tabs/basic.md'
+
+# Push template - basic
+
+A standard notification including a title, subtitle, message body, optional image, icons, link, buttons.<br />Alternate title and body can optionally be specified for when the notification is expanded by the user.
+
+The tables below show all available properties in their respective payloads for FCM (Android) and APNS (iOS).
+<br />
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/templates/basic.md:18]: Any reason for the br tag here? Can't you just do a normal newline?

File: src/pages/resources/reference/push-notifications/templates/carousel.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+---
+title: AEP SDK carousel push template
+description: Carousel push template supported by Adobe Campaign Classic mobile extension.
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/templates/carousel.md:3]: Same comment about metadata here.

File: src/pages/resources/reference/push-notifications/templates/basic.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+---
+title: AEP SDK basic push template
+description: Basic push template supported by Adobe Campaign Classic mobile extension.
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/templates/basic.md:3]: Same comment about metadata here.

File: src/pages/resources/reference/push-notifications/templates/basic.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+- Push
+- Push Template
+- Basic
+---
+
+import Tabs from './tabs/basic.md'
+
+# Push template - basic
+
+A standard notification including a title, subtitle, message body, optional image, icons, link, buttons.<br />Alternate title and body can optionally be specified for when the notification is expanded by the user.
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/templates/basic.md:15]: Same here - any reason for the br tag here?

Comment (calebk1) [reply src/pages/resources/reference/push-notifications/templates/basic.md:15]: "An alternate title and body can be optionally specified when the notification..."

File: src/pages/resources/reference/push-notifications/templates/basic.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+The tables below show all available properties in their respective payloads for FCM (Android) and APNS (iOS).
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/templates/basic.md:17]: "The following tables show all..."

File: src/pages/resources/reference/push-notifications/templates/carousel.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+- Push
+- Push Template
+- Carousel
+---
+
+import Tabs from './tabs/carousel.md'
+
+# Push template - carousel
+
+A notification that shows a series of three (3) to five (5) images that scroll on/off the sides of the notification. <br />If operating in `automatic` mode, images will rotate every five (5) seconds.  If operating in manual mode, the user is required to push a button to advance forwards or backwards through the carousel.
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/templates/carousel.md:15]: Same comment here about the br tag.

Also - for number less than 20, we shouldn't be using the numerical form for it, so I'd just make it "... of three to five images..." and so on for the other numbers.

File: src/pages/resources/reference/push-notifications/templates/index.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
+---
+title: AEP SDK push templates
+description: OOTB push templates supported by the Adobe Campaign Classic mobile SDK extension.
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/templates/index.md:3]: Same metadata comments.

File: src/pages/resources/reference/push-notifications/templates/index.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+keywords:
+- Adobe Campaign Classic
+- Push
+- Push Template
+- Push Templates
+---
+
+# Adobe Experience Platform - push templates
+
+This document outlines each supported out of the box push template.
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/templates/index.md:13]: "This document outlines the push templates that are supported by default."

File: src/pages/resources/reference/push-notifications/templates/tabs/basic.md [hunk 13]
```diff
@@ -91,6 +91,94 @@ Below is a sample of what a payload might look like for a basic notification:
...
+
+### Image specifications
+
+| **Aspect ratios** | **Image size range** | **Supported file types** |
+| :-------- | -----------: | ------: |
+| 1:1, 3:2, 5:4, 4:3 | < 1 MB | PNG, JPG, WebP |
+
+### Text guidelines
+
+| **Type** |**Recommended max<br /># of characters** | **Supports multiple<br />lines?** |
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/templates/tabs/basic.md:106]: Is there a reason for these brs here? I don't think table headers support brs.

File: src/pages/resources/reference/push-notifications/templates/tabs/carousel.md [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+
+### Image specifications
+
+| **Aspect ratios** | **Image size range** | **Supported file types** |
+| :-------- | -----------: | ------: |
+| 1:1, 3:2, 5:4, 4:3 | < 1 MB | PNG, JPG, WebP |
+
+### Text guidelines
+
+| **Type** |**Recommended max<br /># of characters** | **Supports multiple<br />lines?** |
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/templates/tabs/carousel.md:85]: Same comment here with the br tag in the table header.

File: src/pages/resources/reference/push-notifications/templates/tabs/carousel.md [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+
+### Image specifications
+
+| **Aspect ratio** | **Image size range** | **Supported file types** |
+| :-------- | -----------: | ------: |
+| 2:1 (e.g. - 400x200 px) | 50 KB to 1 MB | PNG, JPG, WebP |
+
+### Text guidelines
+
+| **Type** |**Recommended max<br /># of characters** | **Supports multiple<br />lines?** |
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/templates/tabs/carousel.md:176]: Same comment here.

File: src/pages/resources/reference/push-notifications/templates/timer.md [hunk 16]
```diff
@@ -0,0 +1,27 @@
+---
+title: AEP SDK timer push template
+description: Timer push template supported by Adobe Campaign Classic mobile extension.
```
Comment (calebk1) [inline src/pages/resources/reference/push-notifications/templates/timer.md:3]: Metadata.

**Review by calebk1 — CHANGES_REQUESTED** [review]
Comments.

File: src/pages/resources/reference/push-notifications/ios/configure-extension.md [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+- Notification Content Extensions
+- AEPNotificationContent
+- Push
+- Push Template
+- Push Templates
+---
+
+# Configure the Notification Content extension
+
+With the AEPNotificationContent package now available after following the [installation steps](./index.md), the app must now be configured to use the Adobe SDK when the Notification Content extension is called.
```
[the remark this answers is not in the document]
Comment (sbenedicadb) [reply src/pages/resources/reference/push-notifications/ios/configure-extension.md:14]: yes, updated to provide clarity.

File: src/pages/resources/reference/push-notifications/templates/tabs/basic.md (hunk 13 above)
Comment (sbenedicadb) [reply src/pages/resources/reference/push-notifications/templates/tabs/basic.md:106]: normal markdown calculates column width differently - the brs make this table look better on (e.g.) github.

i'll remove them from these docs though as they have no impact - all columns are assigned equal width :(

**Review by calebk1 — APPROVED** [review]
LGTM!

**Pull request merged**